react/utils/metrics.ts

```diff
@@ -0,0 +1,148 @@
+import axios from 'axios'
+
+const ANALYTICS_URL = 'https://rc.vtex.com/api/analytics/schemaless-events'
+
+const GRAPHQL_URL = (accountName: string, workspace?: string) => {
+  if (workspace) {
+    return `https://${workspace}--${accountName}.myvtex.com/_v/private/graphql/v1`
+  }
+
+  return `https://${accountName}.myvtex.com/_v/private/graphql/v1`
+}
+
+type Metric = {
+  name: 'b2b-suite-buyerorg-data'
+  kind: 'create-quote-ui-event'
+  description: 'Create Quotation Action - UI'
+  account: string
+}
+
+type QuoteFieldsMetric = {
+  cost_center_id: string
+  cost_center_name: string
+  buy_org_id: string
+  buy_org_name: string
+  member_id: string
+  member_email: string
+  role: string
+  creation_date: string
+  quote_id: string
+  quote_reference_name: string
+  send_to_sales_rep: boolean
+}
+
+export type SessionProfile = {
+  id: { value: string }
+  email: { value: string }
+}
+
+type SessionResponse = {
+  namespaces: {
+    profile: SessionProfile
+    account: {
+      accountName: { value: string }
+    }
+  }
+}
+
+type MetricsParam = {
+  quoteId: string
+  sessionResponse: SessionResponse
+  workspace: string
+  sendToSalesRep: boolean
+}
+
+type QuoteMetric = Metric & { fields: QuoteFieldsMetric }
+
+const fetchMetricsData = async (
+  accountName: string,
+  workspace: string,
+  quoteId: string,
+  userEmail: string
+) => {
+  const query = JSON.stringify({
+    query: `query GetMetricsData($id: String, $email: String!) {
+      getQuote(id: $id) @context(provider: "vtex.b2b-quotes-graphql") {
+        organization
+        organizationName
+        costCenterName
+        referenceName
+        creatorRole
+        creationDate
+      },
+      getUserByEmail(email: $email) @context(provider: "vtex.storefront-permissions") {
+      costId
+    }
+  }`,
+    variables: { id: quoteId, email: userEmail },
+  })
+
+  const { data } = (
+    await axios.post(GRAPHQL_URL(accountName, workspace), query)
+  ).data
+
```

Comment:
When a GraphQL request fails, the response status is often 200 anyway, depending on the type of the error (mainly because there can be partial errors). So there would be no exception thrown here, but `data` might be undefined, or it might only have part of the information you requested. In those cases the response will contain a field called `errors`, so you can check for that here.

```suggestion
  const { data, errors } = (
    await axios.post(GRAPHQL_URL(accountName, workspace), query)
  ).data

```

More information on graphql errors [here](https://learning.postman.com/open-technologies/blog/graphql-error-handling/).